easybuild/framework/easyconfig/types.py

```diff
@@ -126,18 +126,53 @@ def to_toolchain(tcspec):
         if len(tcspec) == 2:
             res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
         else:
-            raise EasyBuildError("Can not convert list %s to toolchain dict. Expected 2 elements", tcspec)
+            errstr = "Can not convert list %s to name and version dict. Expected 2 elements"
+            raise EasyBuildError(errstr, tcspec)
     else:
-        raise EasyBuildError("Conversion of %s (type %s) to toolchain dict is not supported", tcspec, type(tcspec))
+        errstr = "Conversion of %s (type %s) to name and version dict is not supported"
+        raise EasyBuildError(errstr, tcspec, type(tcspec))
 
     return res
 
 
-# this uses to_toolchain, so it needs to be at the bottom of the module
+def to_dependency(dep):
+    """
+    Convert a dependency dict from the new YAML format {'lib': '1.2.8', 'toolchain': 'gcc, 4.8.2'}
+    to a dependency dictionary
+    """
+    depspec = {}
+    if isinstance(dep, dict):
+        for key, value in dep.items():
+            if key == 'versionsuffix':
+                depspec['versionsuffix'] = value
+            elif key == 'toolchain':
+                depspec['toolchain'] = to_name_version_dict(value)
+            elif 'name' not in depspec:
```

Comment:
I think you need a separate boolean value here to check, something like:

```python
if not found_name_version:
    depspec.update({'name': key, 'version': value})
    found_name_version = True
```

to protect against situations like:

```yaml
- name: foo
  version: 1.2.3
  toolchain: GCC, 4,9.2
  versionsuffix: -blah
```